applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -125,14 +141,24 @@ static bool is_page_clean(const struct flash_area *fa, off_t off, size_t len)
 	return true;
 }
 
+static void terminate_dfu(void)
+{
+	__ASSERT_NO_MSG(flash_area != NULL);
+
+	flash_area_close(flash_area);
+	k_delayed_work_cancel(&dfu_timeout);
+	k_delayed_work_cancel(&background_store);
+	dfu_unlock(MODULE_ID(MODULE));
+	flash_area = NULL;
+	sync_offset = 0;
+}
+
 static void dfu_timeout_handler(struct k_work *work)
 {
 	LOG_WRN("DFU timed out");
 
 	if (flash_area) {
-		dfu_unlock(MODULE_ID(MODULE));
-		flash_area_close(flash_area);
-		flash_area = NULL;
+		terminate_dfu();
```

Comment:
Do we need the `if (flash_area)` check here?

Reply:
Don't think so